quic/s2n-quic/src/provider/mtu.rs

```diff
@@ -0,0 +1,42 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Provides MTU configuration at the connection level.
+
+pub use s2n_quic_core::path::mtu::{Config, Configurator, ConnectionInfo};
+
+pub trait Provider {
+    type Config: 'static + Send + Configurator;
+    type Error: 'static + core::fmt::Display + Send + Sync;
+
+    fn start(self) -> Result<Self::Config, Self::Error>;
+}
+
+pub use default::Provider as Default;
+
+impl_provider_utils!();
+
+impl<T: 'static + Send + Configurator> Provider for T {
+    type Config = T;
+    type Error = core::convert::Infallible;
+
+    fn start(self) -> Result<Self::Config, Self::Error> {
+        Ok(self)
+    }
+}
+
+pub mod default {
```

Comment:
If I'm reading this right, this default provider impl just always returns the default Mtu Config. Shouldn't it instead always return the endpoint mtu config?

Reply:
`mtu::Endpoint` has a default impl for `on_connection` which returns the endpoint_mtu_config. Since mtu::Config doesn't override this, it will return the endpoint config. [code](https://github.com/aws/s2n-quic/blob/5fc9d3cf4773c386b3e93d865bc3e2af02decb81/quic/s2n-quic-core/src/path/mtu.rs#L261-L278)


While here I am going to get rid of the `mod default` and flatten/simplify this (while preserving the current behavior):
```
// This will work since Config already `impl mtu::Endpoint`
pub use Config as Default;
```